### ISSUE TITLE
ipywidgets button click event not work

### ISSUE THREAD

**Issue by JasonZzy0528** [jupyterlab/jupyterlab#1286]
<img width="1280" alt="screen shot 2016-11-23 at 8 21 55 pm" src="https://cloud.githubusercontent.com/assets/10888981/20556302/9f9003c8-b1ba-11e6-87e6-d6707655218c.png">

<img width="1280" alt="screen shot 2016-11-23 at 8 19 38 pm" src="https://cloud.githubusercontent.com/assets/10888981/20556225/34b1a7d2-b1ba-11e6-8e30-3fa3ebd2cf6c.png">

The above is my code. After I click the "Display Chart" button, it should print out "Ay". But it does not. I found the click event is executed after I click button, and its description is changed to 'clicked'.
I am currently using 
1. Python 3.5.2
2. jupyter 1.0.0
3. jupyterlab 0.8.0
4. ipywidgets 6.0.0b2

Is this something wrong with my code or is it jupyerlab's problem?

**Comment by blink1073** [comment]
Thanks for the report, @JasonZzy0528.  I can confirm with a similar setup.  It works as intended in the classic notebook.  I think I heard @jasongrout say that he was aware of this issue and was planning a fix for ipywidgets 6.0 final.  @jasongrout, can you confirm?

**Comment by lmoran** [comment]
Since this is show-stopper for us, could you please suggest a work-around while waiting for 1.0?

**Comment by SylvainCorlay** [comment]
@lmoran we will be looking into it. You can always display a `Text` widget or explicitly use an output widget to display textual information.

**Comment by jasongrout** [comment]
I'm just about ready to merge in a bunch of styling changes and was going to release a new beta. I'll look into this.  @SylvainCorlay - if you're up, I'd love if you'd take a look at https://github.com/ipython/ipywidgets/pull/915

**Comment by lmoran** [comment]
Thanks for the suggestion, @SylvainCorlay . Unfortunately, we use IPyWidgets to build graphic (charts&maps) outputs, not textual ones.

**Comment by jasongrout** [comment]
The relevant issue is https://github.com/jupyterlab/jupyterlab/issues/1254.

Basically, the issue is that we don't know where to put the output generated by the click event. In the classic notebook, we make a guess about where to put it - in the cell where the click happened. This means that if you display the button in several cells, only one of them displays the output. This is kludgy, inconsistent, and doesn't work nearly as well now that widgets are outputs too. And it doesn't work very well with turning widgets into dashboards, etc.

A much more sustainable solution is mentioned in #1254 - if you'd like to see the output, then it is up to you to explicitly capture it with an output widget and show that output widget wherever you like. If you don't capture it, then we should provide a global log of uncaptured outputs that you can see.

Here's the recommended solution now:

```python
import ipywidgets as widgets
button = widgets.Button(description='Display Chart')
out = widgets.Output()
def on_button_clicked(b):
    button.description = 'clicked'
    with out:
        print('Ay')

button.on_click(on_button_clicked)
display(widgets.VBox([button, out]))
```

**Comment by jasongrout** [comment]
Note that displaying charts, `clear_output`, etc. also work within that `with out:` block...

**Comment by lmoran** [comment]
Thanks for the suggested workaround @jasongrout , much appreciated.

**Comment by JasonZzy0528** [comment]
Thanks @jasongrout

**Comment by jasongrout** [comment]
Great! Closing as fixed, and leaving open #1254 to implement a global message log for unclaimed messages.
